### PR TITLE
added an encoding parameter to TextVectorization layer

### PR DESCRIPTION
**Pull request by tonyrubanraj**
PR related to #15800

### CODE REVIEW

File: keras/layers/preprocessing/text_vectorization.py [hunk 1]
```diff
@@ -255,6 +255,7 @@ def __init__(
         idf_weights=None,
         sparse=False,
         ragged=False,
+        encoding=None,
```
Comment (fchollet) [inline keras/layers/preprocessing/text_vectorization.py:258]: The argument would need to be documented in the docstring.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. I noticed the argument is also not documented in `StringLookup` -- please add it to the docstring there as well.

File: keras/layers/preprocessing/string_lookup.py [hunk 2]
```diff
@@ -123,6 +123,8 @@ class StringLookup(index_lookup.IndexLookup):
       sparse: Boolean. Only applicable when `output_mode` is `"multi_hot"`,
         `"count"`, or `"tf_idf"`. If True, returns a `SparseTensor` instead of a
         dense `Tensor`. Defaults to False.
+      encoding: Optional. A string representing the encoding to use. Defaults to
```
Comment (fchollet) [inline keras/layers/preprocessing/string_lookup.py:126]: "the text encoding to use to interpret the input strings"

File: keras/layers/preprocessing/text_vectorization.py [hunk 3]
```diff
@@ -174,6 +174,8 @@ class TextVectorization(base_preprocessing_layer.PreprocessingLayer):
       sparse: Boolean. Only applicable to `"multi_hot"`, `"count"`, and
         `"tf_idf"` output modes. If True, returns a `SparseTensor` instead of a
         dense `Tensor`. Defaults to False.
+      encoding: Optional. A string representing the encoding to use. Defaults to
```
Comment (fchollet) [inline keras/layers/preprocessing/text_vectorization.py:177]: Likewise

Comment (tonyrubanraj) [reply keras/layers/preprocessing/text_vectorization.py]: updated the encoding detail in docstring

File: keras/layers/preprocessing/string_lookup.py (hunk 2 above)
Comment (tonyrubanraj) [reply keras/layers/preprocessing/string_lookup.py]: updated the encoding detail in docstring

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**